### PR TITLE
Fat 1292 c17027

### PR DESCRIPTION
**Pull request by TetianaParanich**
***
https://issues.folio.org/browse/FAT-1292
***
https://foliotest.testrail.io/index.php?/cases/view/17027
***

### CODE REVIEW

File: cypress/integration/data-import/match-on-location.spec.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    // NewJobProfile.saveAndClose();
+    // JobProfiles.checkJobProfilePresented(updateJobProfile.profileName);
+
+    // // upload a marc file
+    // cy.visit(TopMenu.dataImportPath);
+    // DataImport.uploadFile(editedFileNameRev1, fileNameForProtect);
+    // JobProfiles.searchJobProfileForImport(updateJobProfile.profileName);
+    // JobProfiles.runImportFile(fileNameForProtect);
+    // Logs.checkStatusOfJobProfile('Completed');
+    // Logs.openFileDetails(fileNameForProtect);
```
Comment (penfriend) [inline cypress/integration/data-import/match-on-location.spec.js:141]: please remove or uncomment this code

Comment (TetianaParanich) [reply cypress/integration/data-import/match-on-location.spec.js]: Done

**Review by MaksimTerekhov — APPROVED** [review]

File: cypress/integration/data-import/match-on-location.spec.js [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+      });
+    });
+    DataImportSettingsJobProfiles.createJobProfileApi(testData.jobProfileForCreate)
+      .then((bodyWithjobProfile) => {
+        testData.jobProfileForCreate.id = bodyWithjobProfile.body.id;
+      });
+
+    // upload a marc file for creating of the new instance, holding and item
+    cy.visit(TopMenu.dataImportPath);
+    DataImport.uploadFile('matchOnLocation.mrc', marcFileForCreate);
```
Comment (penfriend) [inline cypress/integration/data-import/match-on-location.spec.js:259]: A fixture file could not be found at any of the following paths:

    > cypress\fixtures\matchOnLocation.mrc
    > cypress\fixtures\matchOnLocation.mrc.[ext]

File: cypress/integration/data-import/match-on-location.spec.js [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+import FieldMappingProfiles from '../../support/fragments/data_import/mapping_profiles/fieldMappingProfiles';
+import ActionProfiles from '../../support/fragments/data_import/action_profiles/actionProfiles';
+import JobProfiles from '../../support/fragments/data_import/job_profiles/jobProfiles';
+import TopMenu from '../../support/fragments/topMenu';
+import DataImport from '../../support/fragments/data_import/dataImport';
+import FileDetails from '../../support/fragments/data_import/logs/fileDetails';
+import Logs from '../../support/fragments/data_import/logs/logs';
+import DataImportSettingsJobProfiles from '../../support/fragments/settings/dataImport/dataImportSettingsJobProfiles';
+import SearchInventory from '../../support/fragments/data_import/searchInventory';
+import InventoryInstance from '../../support/fragments/inventory/inventoryInstance';
```
Comment (penfriend) [inline cypress/integration/data-import/match-on-location.spec.js:18]: 'InventoryInstance' is defined but never used. Allowed unused vars must match /React/u

File: cypress/integration/data-import/match-on-location.spec.js [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+import NewMappingProfile from '../../support/fragments/data_import/mapping_profiles/newMappingProfile';
+
+describe('ui-data-import: Match on location', () => {
+  const permanentLocation = 'Annex (KU/CC/DI/A)';
+  const recordType = 'MARC_BIBLIOGRAPHIC';
+  const instanceTitle = 'Anglo-Saxon manuscripts in microfiche facsimile Volume 25 Corpus Christi College, Cambridge II, MSS 12, 144, 162, 178, 188, 198, 265, 285, 322, 326, 449 microform A. N. Doane (editor and director), Matthew T. Hussey (associate editor), Phillip Pulsiano (founding editor)';
```
Comment (penfriend) [inline cypress/integration/data-import/match-on-location.spec.js:24]: 'instanceTitle' is assigned a value but never used. Allowed unused vars must match /React/u

File: cypress/e2e/data-import/edi-import-file-for-create-invoice-record.cy.js [hunk 5]
```diff
@@ -37,17 +37,17 @@ describe('ui-data-import: EDIFACT file import with creating of new invoice recor
...
-    // clean up generated profiles
-    JobProfiles.deleteJobProfile(jobProfileName);
-    ActionProfiles.deleteActionProfile(actionProfileName);
-    FieldMappingProfiles.deleteFieldMappingProfile(mappingProfileName);
-  });
+  //   // clean up generated profiles
+  //   JobProfiles.deleteJobProfile(jobProfileName);
+  //   ActionProfiles.deleteActionProfile(actionProfileName);
+  //   FieldMappingProfiles.deleteFieldMappingProfile(mappingProfileName);
+  // });
```
Comment (re-jas) [inline cypress/e2e/data-import/edi-import-file-for-create-invoice-record.cy.js:50]: there are shouldn't be comments in pr

File: cypress/e2e/data-import/edi-import-file-for-create-invoice-record.cy.js [hunk 6]
```diff
@@ -37,17 +37,17 @@ describe('ui-data-import: EDIFACT file import with creating of new invoice recor
     DataImport.checkUploadState();
   });
 
-  after(() => {
-    cy.getInvoiceIdApi({ query: `vendorInvoiceNo="${FileDetails.invoiceNumberFromEdifactFile}"` })
-      .then(id => cy.deleteInvoiceFromStorageApi(id));
-    DataImport.checkUploadState();
-    Users.deleteViaApi(user.userId);
+  // after(() => {
```
Comment (re-jas) [inline cypress/e2e/data-import/edi-import-file-for-create-invoice-record.cy.js:40]: 'Users' is defined but never used. Allowed unused vars must match /React/u

File: cypress/integration/data-import/match-on-location.spec.js (hunk 2 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/match-on-location.spec.js]: Done

File: cypress/integration/data-import/match-on-location.spec.js [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+    // upload a marc file for creating of the new instance, holding and item
+    cy.visit(TopMenu.dataImportPath);
+    DataImport.uploadFile('matchOnLocation.mrc', marcFileForCreate);
+    JobProfiles.searchJobProfileForImport(testData.jobProfileForCreate.profile.name);
+    JobProfiles.runImportFile(marcFileForCreate);
+    Logs.openFileDetails(marcFileForCreate);
+    FileDetails.checkItemsStatusesInResultList(0, [FileDetails.status.created, FileDetails.status.created, FileDetails.status.created, FileDetails.status.created]);
+    // FileDetails.checkItemsStatusesInResultList(1, [FileDetails.status.created, FileDetails.status.created, FileDetails.status.created, FileDetails.status.created]);
+    // FileDetails.checkItemsStatusesInResultList(2, [FileDetails.status.created, FileDetails.status.created, FileDetails.status.created, FileDetails.status.created]);
+    //FileDetails.checkItemsQuantityInSummaryTable(0, '3');
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/data-import/match-on-location.spec.js]: Done

File: cypress/integration/data-import/match-on-location.spec.js [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+  //   FieldMappingProfiles.deleteFieldMappingProfile(itemMappingProfileName);
+  //   FieldMappingProfiles.deleteFieldMappingProfile(itemUpdateMappingProfile);
+
+  //   cy.getInstance({ limit: 1, expandAll: true, query: `"title"=="${instanceTitle}"` })
+  //     .then((instance) => {
+  //       cy.deleteItem(instance.items[0].id);
+  //       cy.deleteHoldingRecordViaApi(instance.holdings[0].id);
+  //       InventoryInstance.deleteInstanceViaApi(instance.id);
+  //     });
+  // });
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/data-import/match-on-location.spec.js]: Done

File: cypress/integration/data-import/match-on-location.spec.js [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+    // create Field mapping profiles
+    cy.visit(SettingsMenu.mappingProfilePath);
+
+    // cy.visit(SettingsMenu.mappingProfilePath);
+    // FieldMappingProfiles.createHoldingsMappingProfileWithNotes(holdingsUpdateMappingProfile, noteForHoldingsMappingProfile);
+    // FieldMappingProfiles.checkMappingProfilePresented(holdingsUpdateMappingProfile.name);
+
+    // FieldMappingProfiles.createItemMappingProfileWithNotes(itemUpdateMappingProfile, noteForItemMappingProfile);
+    // FieldMappingProfiles.checkMappingProfilePresented(itemUpdateMappingProfile.name);
+
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/data-import/match-on-location.spec.js]: Done

File: cypress/integration/data-import/match-on-location.spec.js [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+    // upload a marc file
+    cy.visit(TopMenu.dataImportPath);
+    DataImport.uploadFile(editedMarcFileName, fileNameAfterUpdate);
+    JobProfiles.searchJobProfileForImport(jobProfileNameForUpdate.profileName);
+    JobProfiles.runImportFile(fileNameAfterUpdate);
+    Logs.checkStatusOfJobProfile('Completed');
+    Logs.openFileDetails(fileNameAfterUpdate);
+
+    // FileDetails.checkItemsQuantityInSummaryTable(0, '1');
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/data-import/match-on-location.spec.js]: Done

File: cypress/integration/data-import/match-on-location.spec.js (hunk 3 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/match-on-location.spec.js]: Done

File: cypress/integration/data-import/match-on-location.spec.js (hunk 4 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/match-on-location.spec.js]: Done

File: cypress/e2e/data-import/edi-import-file-for-create-invoice-record.cy.js (hunk 5 above)
Comment (re-jas) [inline cypress/e2e/data-import/edi-import-file-for-create-invoice-record.cy.js:50]: there are shouldn't be comments

**Review by re-jas — CHANGES_REQUESTED** [review]

File: cypress/support/fragments/data_import/mapping_profiles/newFieldMappingProfile.js [hunk 11]
```diff
@@ -14,7 +14,6 @@ import getRandomPostfix from '../../../utils/stringTools';
 
 const saveButton = Button('Save as profile & Close');
 const organizationModal = Modal('Select Organization');
-const adminNoteFieldName = 'profile.mappingDetails.mappingFields[9].repeatableFieldAction';
 const marcBib = 'MARC Bibliographic';
```
Comment (re-jas) [inline cypress/support/fragments/data_import/mapping_profiles/newFieldMappingProfile.js:17]: 'marcBib' is assigned a value but never used. Allowed unused vars must match /React/u

it seems to me that you initially created the variable "marcBib" in order to pass its value to the properties of the "incomingRecordType" and "folioRecordTypeValue" objects, but you did not do this, so please delete the variable if you do not need it

**Review by re-jas — CHANGES_REQUESTED** [review]

**Review by re-jas — APPROVED** [review]

**Pull request merged**